grpc/bcodes.go

```diff
@@ -62,10 +67,25 @@ func errorToCode(err error) codes.Code {
 	}
 }
 
-func wrapError(err error) error {
+// wrapError wraps the internal error types we use for transport across the gRPC
```

Comment:
I feel like this is still not clear enough. Can you add "and appends an appropriate errortype to the gRPC trailer via the provided context" at the top?